python/catkin/builder.py

```diff
@@ -1217,7 +1221,7 @@ def get_package_names_with_recursive_dependencies(packages, pkg_names):
                 pkg.run_depends +
                 (pkg.test_depends if pkg.package_format > 1 else [])
             )
-            for dep in [dep.name for dep in deps_to_iterate_over]:
+            for dep in [dep.name for dep in deps_to_iterate_over if d is not False]:
```

Comment:
@dirk-thomas After this pull request, I noticed the `catkin` now throws exception as below:

```
Traceback (most recent call last):
Base path: c:\catkin_ws
  File "src\catkin\bin\catkin_make_isolated.py", line 165, in <module>
Source space: c:\catkin_ws\src
    main()
Build space: c:\catkin_ws\build_isolated
  File "src\catkin\bin\catkin_make_isolated.py", line 161, in main
Devel space: c:\catkin_ws\devel_isolated
    override_build_tool_check=opts.override_build_tool_check,
Install space: c:\opt\ros\melodic\x64
  File "src\catkin\bin\..\python\catkin\builder.py", line 963, in build_workspace_isolated
Additional CMake Arguments: -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_PREFIX_PATH=C:/opt/ros/melodic/x64;C:/opt/rosdeps/x64 -DCMAKE_VERBOSE_MAKEFILE=ON -DPYTHON_VERSION=2.7 -DPYTHON_EXECUTABLE=C:\opt\python27amd64\python.exe -DPYTHON_LIBRARIES=C:\opt\python27amd64\Libs -DCATKIN_SKIP_TESTING=ON
    whitelist_pkg_names = get_package_names_with_recursive_dependencies(packages, only_pkg_with_deps)
  File "src\catkin\bin\..\python\catkin\builder.py", line 1224, in get_package_names_with_recursive_dependencies
    for dep in [dep.name for dep in deps_to_iterate_over if d is not False]:
NameError: global name 'd' is not defined
```

Is it a typo for the usage of `d`?

Reply:
Also reported in #1039.